sensor/tests/resource/imagescan/imagescan_test.go

```diff
@@ -0,0 +1,153 @@
+			},
+			EventSource:   0,
+			PolicyVersion: "1.1",
+			PolicySections: []*storage.PolicySection{
+				{
+					SectionName: "",
+					PolicyGroups: []*storage.PolicyGroup{
+						{
+							FieldName:       "Image Component",
+							BooleanOperator: storage.BooleanOperator_OR,
+							Negate:          false,
+							Values: []*storage.PolicyValue{
+								{
+									Value: "rpm|microdnf|dnf|yum=",
+								},
+							},
+						},
+					},
+				},
+			},
+			CriteriaLocked:     true,
+			MitreVectorsLocked: true,
+			IsDefault:          true,
+		},
+	}
+)
+
+type ImageScanSuite struct {
+	testContext *resource.TestContext
+	suite.Suite
+}
+
+func Test_ImageScan(t *testing.T) {
+	suite.Run(t, new(ImageScanSuite))
+}
+
+func (s *ImageScanSuite) SetupSuite() {
+	testContext, err := resource.NewContextWithConfig(s.T(), resource.CentralConfig{
+		InitialSystemPolicies: Policies,
+	})
+	s.Require().NoError(err)
+	s.testContext = testContext
+}
+
+func (s *ImageScanSuite) TearDownTest() {
+	s.testContext.GetFakeCentral().ClearReceivedBuffer()
+}
+
+func (s *ImageScanSuite) Test_AlertsUpdatedOnImageUpdate() {
+	s.testContext.RunTest(
+		resource.WithResources([]resource.K8sResourceInfo{Pod}),
+		resource.WithTestCase(func(t *testing.T, tc *resource.TestContext, resource map[string]k8s.Object) {
+			var image *storage.ContainerImage
+			// Image should be received by central
+			tc.LastDeploymentStateWithTimeout("myapp", func(dp *storage.Deployment, _ central.ResourceAction) error {
+				if len(dp.GetContainers()) != 1 {
+					return errors.Errorf("expected 1 container found %d", len(dp.GetContainers()))
+				}
+
+				if dp.GetContainers()[0].GetImage().GetId() == "" {
+					return errors.New("image ID should not be empty")
+				}
+
+				image = dp.GetContainers()[0].GetImage()
+				return nil
+			}, "myapp should have started the container and have an imageID", 2*time.Minute)
+
+			// There should be no violation yet, because there are no components provided for this image
+			tc.LastViolationState("myapp", func(result *central.AlertResults) error {
+				if checkIfAlertsHaveViolation(result, Policies[0].GetName()) {
+					return errors.Errorf("violation found for deployment")
+				}
+				return nil
+			}, "Should not have violation")
+
+			tc.GetFakeCentral().StubMessage(&central.MsgToSensor{
+				Msg: &central.MsgToSensor_UpdatedImage{
+					UpdatedImage: &storage.Image{
+						Id:    image.GetId(),
+						Name:  image.GetName(),
+						Names: []*storage.ImageName{image.GetName()},
+						Scan: &storage.ImageScan{
+							ScannerVersion: "2.0",
+							Components: []*storage.EmbeddedImageScanComponent{
+								{
+									Name:    "rpm",
+									Version: "3.2.1",
+								},
+							},
+						},
+					},
+				},
+			})
+
+			tc.GetFakeCentral().StubMessage(&central.MsgToSensor{
+				Msg: &central.MsgToSensor_ReprocessDeployments{
+					ReprocessDeployments: &central.ReprocessDeployments{},
+				},
+			})
```

Comment:
If we decide to trigger affected deployments on `UpdatedImage`, then the `ReprocessDeployments` shouldn't be needed here. I think, unless there's a guarantee that after `UpdatedImage` we are going to receive `ReprocessDeployments`, we should trigger affected deployments on `UpdatedImage`, thus I think the tests should test this without sending a `ReprocessDeployments`.

Reply:
See comment below